Calol2 online dqm 91x

ashtipliyski opened the pull request:
Adds event by event comparisons by objects generated by Layer 2 of the calorimeter trigger (CALOL2) which is responsible for jets, e/g, tau and sums.

ashtipliyski committed:
Add initial version of histograms for new sum quantities

ashtipliyski committed:
Add labels to most calo layer 2 histograms

ashtipliyski committed:
Add initial version of tower count histogram to calolayer2

ashtipliyski committed:
Add labels to min bias feature bit plots in calo layer 2

ashtipliyski committed:
Add new quantities to data-emulator comparisons for jets and sums

Conflicts:
	DQM/L1TMonitorClient/interface/L1TStage2CaloLayer2DEClient.h
	DQM/L1TMonitorClient/src/L1TStage2CaloLayer2DEClient.cc

ashtipliyski committed:
Update labels for calo layer 2 objects (jets, sums, e/g and tau)

Conflicts:
	DQM/L1TMonitorClient/src/L1TStage2CaloLayer2DEClient.cc

Commit:
Add initial version of DQM module (cc, header and python config) for CALOL2 online DQM

Commit:
Add CALOL2 online DQM event by event comparison module to main L1T dqm sequence

Commit:
Add summary plots for all objects and refactor them to show relative fractions instead of absolute values. Optimised the summary for sums

Commit:
Refactor the summary plots in CALOL2 online DQM to show fracitonal counts.

cmsbuild commented:
A new Pull Request was created by  @ashtipliyski (Antoni Shtipliyski) for master.

It involves the following packages:

DQM/L1TMonitor


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@kreczko this is something you requested to watch as well.
@muzaffar, @davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

dmitrijus commented:
Please don't print anything to std:cout!

Commit:
Refactor the comparison summary to use endRun(), expand documentation and add histograms for problematic sums.

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

Commit:
Refactor L1TdeStage2CaloLayer2 DQM module to use endOfLuminosityBlock() instead of endRun() and change L1T2016EMU -> L1TEMU

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

Commit:
Revert L1TStage2EMU DQM client to online configuration with L1T2016->L1T

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

ashtipliyski commented:
After deployment at the DQM online playback machine it was noticed that the DQM module was not behaving as expected. To remedy this, the module code was refactored to use endOfLuminosityBlock() instead of endRun() and the module configuration file was modified to replace L1T2016EMU with L1TEMU (already implemented by other L1T PRs deployed at playback machine).

dmitrijus commented:
So, I guess there will be a follow-up - implementing endOfLuminosityBlock

ashtipliyski commented:
Hi @dmitrijus,

Yes, after our discussion in the CMS control room on Saturday I am currently implementing the "summary" functionality using the DQMEDHarvester::dqmEndLuminosityBlock() method as per your recommendation. Will update the PR as soon as possible.

Cheers

Commit:
Add initial version of fractional summary plot via DQMHarvester

Commit:
Restructure event by event comparison plots for CALOL2 and add a histogram to summarise the cause of differences

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0d233935e7f9fbbf4025cb009cd859237850ff0d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/19987/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/19987/summary.html

Comparison Summary:
 - You potentially added 180 lines to the logs
 - Reco comparison results: 3325 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1833901
 - DQMHistoTests: Total failures: 47749
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1785972
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

Commit:
Investigate the cause of segfault when MET Phi data, emulator histograms are divided in CALOL2 emulator DQM.

ashtipliyski commented:
please test

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3cccf678ae0fcc4e8e2de53702a29f08780c999e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/20069/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/20069/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1833024
 - DQMHistoTests: Total failures: 32772
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1800072
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

davidlange6 commented:
hi- is this PR obsolete now?

dmitrijus commented:
@davidlange6 It is used in online DQM. Should be merged.

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

davidlange6 commented:
please test

(the tests are ancient due to the slow dqm review)

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3cccf678ae0fcc4e8e2de53702a29f08780c999e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/21371/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/92ee79880f3dc245397a00e6acced67d57d310cc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/21371/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/21371/git-merge-result

cmsbuild commented:
Comparison job queued.

davidlange6 commented on DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:
```diff
@@ -0,0 +1,1234 @@
+  if (dataCol->size() != emulCol->size()) {
+
+    if (dataCol->size() < emulCol->size()) {
+      if (dataCol->size() < 1) {
```
hi @ashtipliyski - this can be if ( dataCol->empty() ) return false

davidlange6 commented on DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:
```diff
@@ -0,0 +1,1234 @@
+      // jet Et mismatch
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+        etGood = false;
+ 	eventGood = false;
```
please fix the indentation here

ashtipliyski replied:
Hi @davidlange6, both indentation and comment in this location have been updated. To confirm, please have a look at the whole file individually. Other comments are welcome.

Thanks

davidlange6 commented on DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:
```diff
@@ -0,0 +1,1234 @@
+
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+	eventGood = false;
+	ettEmData->getTH1F()->Fill(dataIt->hwPt());
```
essentially none of the getTH1F() are needed - the MonitorElement has a Fill method to call directly.

davidlange6 commented on DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:
```diff
@@ -0,0 +1,1234 @@
+	summaryHist->Fill(SUMGOOD_S);
+	objSummaryHist->Fill(MHTSUMGOOD);
+      } else {
+       	mhtHFEtData->getTH1F()->Fill(dataIt->hwPt());
```
indentation problem

davidlange6 commented on DQM/L1TMonitorClient/src/L1TStage2CaloLayer2DEClient.cc:
```diff
@@ -1,48 +1,424 @@
+
+    double evtRatio = 0, jetRatio = 0, egRatio = 0, tauRatio = 0, sumRatio = 0;
+
+    totalEvents = hist->GetBinContent(1);
```
please reformat this to combine the declaration and setting of each variable

double totalEvents = hist->GetBinContent(1);

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/21371/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2012315
 - DQMHistoTests: Total failures: 109
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2012040
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

davidlange6 commented:
-1
pending answers to previous comments
(in addition all of the getTH1F()s here are not needed -please remove them and fill histograms via dqm interface directly)

ashtipliyski committed:
Implement review comments and add bug-fixes to summary plots.

- remove references to TH1F()
- remove debugging couts
- mitigate bugs in population of summary histograms

cmsbuild commented:
Pull request #18397 was updated. @cmsbuild, @smuzaffar, @mrodozov can you please check and sign again.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
Pull request #18397 was updated.  can you please check and sign again.

cmsbuild commented:
Pull request #18397 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

ashtipliyski commented:
Apologies for the messy commit from 2h ago. I had created a CMSSW area based off a version that was ahead from 9.0.2 (what this PR was originally based on) and that brought a lot of irrelevant commits with it. 

I believe that it should all be in order now.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2a4ea14f627806ff9f308b5d01e99b686d238fbc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22155/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/e4b262ab759cd9d703347718da74e34ae74c660c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22155/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22155/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22155/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2646495
 - DQMHistoTests: Total failures: 773
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2645541
 - DQMHistoTests: Total skipped: 181
 - DQMHistoTests: Total Missing objects: 0
 - Checked 102 log files, 14 edm output root files, 25 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

Commit:
Apropriately set the default values of summary bins when no objects are found

cmsbuild commented:
Pull request #18397 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-18397/52

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-18397/52/git-diff.patch

In future, you can run `scram build code-checks` to apply code checks

ashtipliyski commented:
Latest update includes changes to the default values of the summary histograms the need for which became apparent only after the PR was run in production DQM and an edge case was encountered (data taking with cosmics and no reconstructed tau leptons).

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: e1b29bee2ee8964bbcdf3c212a4f9657fcf05e70
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22276/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/8ef7e73e06eaf3499e183006f0ef93df28fa0c09
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22276/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22276/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22276/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2713569
 - DQMHistoTests: Total failures: 261
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2713119
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

ashtipliyski commented:
@smuzaffar What is the significance of the code-checks-rejected label?

smuzaffar commented:
@ashtipliyski , this label shows the state of code-checks tests https://github.com/cms-sw/cmssw/pull/18397#issuecomment-321784778

you can apply the patch give in the above comment

smuzaffar commented:
code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-18397/273

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-18397/273/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-18397/273/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

ashtipliyski commented:
A lot of the changes suggested by the patch relate to modules outside of the focus of this PR and have broader implications for DQM. @dmitrijus, do you agree with the suggestions in the patch generated from cmsbuild?

smuzaffar commented:
@ashtipliyski check the changes suggested byt he latest run https://cmssdt.cern.ch/SDT/code-checks/PR-18397/273/git-diff.patch

davidlange6 commented on DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:
```diff
@@ -0,0 +1,1234 @@
+      // jet Et mismatch
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+        etGood = false;
+ 	eventGood = false;
```
hi @ashtipliyski - it looks like this comment should still be fixed.

Commit:
Fix typo in referencing MHT/MHTHF phi values for CALOL2 event by event comparisons. Update inline comments.

Commit:
Fix typo in referencing MHT/MHTHF phi values for CALOL2 event by event comparisons. Update inline comments.

Commit:
Apply patch with changes suggested by cmsbuild.

ashtipliyski commented:
Fixed a typo in the histogram population script and applied changes suggested by @davidlange6 and @smuzaffar.

ashtipliyski commented:
-code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #18397 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-18397/330

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: cd99957cdd2b6f2f9e9630359957d9af0f352048
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22615/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/94fab3e019e00215c7cc5373d84b920601f2aef5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22615/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22615/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18397/22615/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2656041
 - DQMHistoTests: Total failures: 210
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2655642
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

davidlange6 commented:
+1

smuzaffar commented:
@ashtipliyski , can you please provide a fix to avoid the following compilation errors for gcc7 IBs
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-09-04-2300/DQM/L1TMonitor
```
DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc: In member function 'virtual void L1TdeStage2CaloLayer2::analyze(const edm::Event&, const edm::EventSetup&)':
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:341:43: error: variable 'dataEGIt' set but not used [-Werror=unused-but-set-variable]
    l1t::EGammaBxCollection::const_iterator dataEGIt = egDataCol->begin(currBx);
                                           ^~~~~~~~
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:342:43: error: variable 'emulEGIt' set but not used [-Werror=unused-but-set-variable]
    l1t::EGammaBxCollection::const_iterator emulEGIt = egEmulCol->begin(currBx);
                                           ^~~~~~~~
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:344:40: error: variable 'dataTauIt' set but not used [-Werror=unused-but-set-variable]
    l1t::TauBxCollection::const_iterator dataTauIt = tauDataCol->begin(currBx);
                                        ^~~~~~~~~
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:345:40: error: variable 'emulTauIt' set but not used [-Werror=unused-but-set-variable]
    l1t::TauBxCollection::const_iterator emulTauIt = tauEmulCol->begin(currBx);
                                        ^~~~~~~~~
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:347:42: error: variable 'dataSumIt' set but not used [-Werror=unused-but-set-variable]
    l1t::EtSumBxCollection::const_iterator dataSumIt = sumDataCol->begin(currBx);
                                          ^~~~~~~~~
  DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc:348:42: error: variable 'emulSumIt' set but not used [-Werror=unused-but-set-variable]
    l1t::EtSumBxCollection::const_iterator emulSumIt = sumEmulCol->begin(currBx);
````

ashtipliyski commented:
Hi @smuzaffar, code to remove unused variables can be found here https://github.com/ashtipliyski/cmssw/commit/d04e83dbe378d81f652bb0f8c5498996c493ace8.

smuzaffar commented:
Ok thanks , I have open a PR https://github.com/cms-sw/cmssw/pull/20390
